plugins/rplanners/rrt.h

```diff
@@ -456,6 +457,13 @@ Some python code to display data::\n\
             // extend A
             ExtendType et = TreeA->Extend(_sampleConfig, iConnectedA);
 
+            if (et == ET_Failed && IS_DEBUGLEVEL(Level_Verbose)) {
+                std::vector<dReal> robotJointValues;
+                this->_parameters->_getstatefn(robotJointValues); // Making the assumption that _getstatefn is shared throughout planning, e.g. shares same _getstatefn as DynamicsCollisionConstraint checker
+                planningstatus.vRobotJointValues.push_back(robotJointValues);                           // Is this slow?
+                planningstatus.vCollidingBodies.push_back(this->_treeForward.GetCollisionBodies());    // Is this slow?
```

Comment:
1. Colliding link will be more useful than colliding bodies.
2. I think you need to do the count collection at this level instead of letting the caller do it.
3. Saving `std::pair<KinBody::LinkConstPtr, KinBody::LinkConstPtr>` instead of `std::string` will probably be better, caller can figure out the body and the link.